gateway-server/src/main/java/org/apache/knox/gateway/config/impl/GatewayConfigImpl.java

```diff
@@ -235,8 +238,35 @@
       "NAMENODE", "JOBTRACKER", "WEBHDFS", "WEBHCAT",
       "OOZIE", "WEBHBASE", "HIVE", "RESOURCEMANAGER");
 
-  public GatewayConfigImpl() {
+  public GatewayConfigImpl() throws GatewayConfigurationException {
     init();
+    validate();
+  }
+
+  private void validate() throws GatewayConfigurationException {
```

Comment:
It seems like this would be better done right before it is used.  Like in `org.apache.knox.gateway.GatewayServer#configureKerberosSecurity`.  I guess for code reuse, this might be ok. However, the KnoxCLI will unnecessarily fail if all is not setup properly for the Gateway.

Reply:
Thanks, fixed!